### PR TITLE
Event tool support

### PR DESCRIPTION
**Pull request by matt-rich**

### CODE REVIEW

File: event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/JdbcEventRepository.java [hunk 1]
```diff
@@ -90,9 +94,20 @@ public long getCurrentSequenceIdForStream(final UUID streamId) {
...
 
+    @Override
+    public Stream<Stream<JsonEnvelope>> getStreamOfAllActiveEventStreams() {
+        final Stream<UUID> streamIds = eventStreamJdbcRepository.findActive().map(es -> es.getStreamId());
+        return streamIds
+                .map(id -> {
+                    final Stream<Event> eventStream = eventJdbcRepository.findByStreamIdOrderBySequenceIdAsc(id);
+                    streamIds.onClose(eventStream::close);
+                    return eventStream.map(eventConverter::envelopeOf);
+                });
```
Comment (mapingo) [inline event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/JdbcEventRepository.java]: The code above is duplicated in another method in the class...could be a method

File: framework-api/framework-api-event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceConstants.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.services.eventsourcing.source.core;
+
+public abstract class EventSourceConstants {
```
Comment (mapingo) [inline framework-api/framework-api-event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceConstants.java]: Wouldn't this be better as a final class with a private constructor?

File: event-sourcing/event-source/pom.xml [hunk 3]
```diff
@@ -41,6 +41,11 @@
             <artifactId>messaging-core</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>core</artifactId>
```
Comment (mapingo) [inline event-sourcing/event-source/pom.xml]: Is the core and the messaging-core dependencies required?

File: event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/JdbcEventRepository.java (hunk 1 above)
Comment (matt-rich) [reply event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/JdbcEventRepository.java]: Done

File: core/src/test/java/uk/gov/justice/services/core/enveloper/DefaultEnveloperTest.java [hunk 4]
```diff
@@ -164,6 +164,24 @@ public void shouldRemoveStreamMetadata() throws JsonProcessingException {
         assertThat(event.metadata().version(), is(empty()));
     }
 
+    @Test
+    public void shouldRemoveStreamMetadataWithName() throws JsonProcessingException {
+        enveloper.register(new EventFoundEvent(TestEvent.class, TEST_EVENT_NAME));
+
+        JsonEnvelope event = enveloper.withMetadataFrom(
```
[the remark this answers is not in the document]
Comment (matt-rich) [reply core/src/test/java/uk/gov/justice/services/core/enveloper/DefaultEnveloperTest.java]: Done

File: event-sourcing/event-source/pom.xml (hunk 3 above)
Comment (matt-rich) [reply event-sourcing/event-source/pom.xml]: Yea, we now need the Enveloper.

File: framework-api/framework-api-event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceConstants.java (hunk 2 above)
Comment (matt-rich) [reply framework-api/framework-api-event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceConstants.java]: Ok, done.

**Pull request merged**